### PR TITLE
[ServiceNow] Mirror In Closing Option

### PR DESCRIPTION
**Pull request by samuelFain**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
relates: https://jira-hq.paloaltonetworks.local/browse/CIAC-5016

## Description
Implemented Mirror In Closing Option in SNOW.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/ServiceNow/ReleaseNotes/2_5_7.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+- Added a Mirror In Closing Option in ServiceNow.
+- Updated the Docker image to: *demisto/python3:3.10.9.45313*.
+
+#### Layouts
+##### ServiceNow Ticket
+- Updated the ServiceNow Ticket Layout.
+
+#### Mappers
+##### ServiceNow - Incoming Mapper
+- Updated the ServiceNow Incoming Mapper.
```
Comment (Shellyber) [inline Packs/ServiceNow/ReleaseNotes/2_5_7.md:13]: Please specify the field/s you've added

File: Packs/ServiceNow/ReleaseNotes/2_5_7.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+
+#### Integrations
+##### ServiceNow v2
+- Added a Mirror In Closing Option in ServiceNow.
+- Updated the Docker image to: *demisto/python3:3.10.9.45313*.
+
+#### Layouts
+##### ServiceNow Ticket
+- Updated the ServiceNow Ticket Layout.
```
Comment (Shellyber) [inline Packs/ServiceNow/ReleaseNotes/2_5_7.md:9]: Please specify the fields you've added

File: Packs/ServiceNow/ReleaseNotes/2_5_7.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+
+#### Integrations
+##### ServiceNow v2
+- Added a Mirror In Closing Option in ServiceNow.
```
Comment (Shellyber) [inline Packs/ServiceNow/ReleaseNotes/2_5_7.md:4]:
```suggestion
- Added the **close_incident_multiple_options** integration param to replace **close_incident** integration param. Now it is possible to change the incident-closing method. The options are 'closed' or 'resolved'
```

**Review by Shellyber — CHANGES_REQUESTED** [review]
Nice work!
Please add the new param to the README as well with good explanation

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 4]
```diff
@@ -2446,37 +2446,46 @@ def get_remote_data_command(client: Client, args: Dict[str, Any], params: Dict)
...
-                'Type': EntryType.NOTE,
-                'Contents': {
-                    'dbotIncidentClose': True,
-                    'closeNotes': f'From ServiceNow: {ticket.get("close_notes")}',
-                    'closeReason': converts_state_close_reason(ticket.get("state"))
-                },
-                'ContentsFormat': EntryFormat.JSON
-            })
+    # Handle closing ticket/incident in XSOAR as chosen by close_incident_multiple_options parameter
+    if ticket.get('closed_at') and params.get('close_incident_multiple_options') == 'closed':
```
Comment (Shellyber) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:2450]: You can merge this two if's together with an or between them, and then continue the function as is, no need to split it and create a new function for it

File: Packs/ServiceNow/ReleaseNotes/2_5_7.md (hunk 1 above)
Comment (samuelFain) [reply Packs/ServiceNow/ReleaseNotes/2_5_7.md]: Due to the PR split, these modifications have been undone, making them irrelevant for these release notes.

File: Packs/ServiceNow/ReleaseNotes/2_5_7.md (hunk 2 above)
Comment (samuelFain) [reply Packs/ServiceNow/ReleaseNotes/2_5_7.md]: Due to the PR split, these modifications have been undone, making them irrelevant for these release notes.

File: Packs/ServiceNow/ReleaseNotes/2_5_7.md (hunk 3 above)
Comment (samuelFain) [reply Packs/ServiceNow/ReleaseNotes/2_5_7.md]: Done 👍

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py (hunk 4 above)
Comment (samuelFain) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py]: Done.

**Comment by samuelFain** [comment]
@Shellyber 
All of the comments have been addressed.
Please review at the changes I made to README.me.
Thanks!

File: Packs/ServiceNow/Integrations/ServiceNowv2/README.md [hunk 5]
```diff
@@ -38,9 +38,7 @@ These scripts are wrapped around the incident table, so to wrap them around anot
    2. Under **Classifier**, select ServiceNow Classifier.
    3. Under **Mapper (incoming)**, select ServiceNow - Incoming Mapper.
    4. Under **Mapper (outgoing)**, select ServiceNow - Outgoing Mapper.
-   5. To enable mirroring to close a ticket in ServiceNow, select the **Close Mirrored XSOAR Incident** checkbox.
-
-        ![image](https://raw.githubusercontent.com/demisto/content/75395ba6d9118bc3a5a399a31d95de4dc27f0911/Packs/ServiceNow/Integrations/ServiceNowv2/doc_files/closing-mirror-xsoar.png)
+   5. To enable mirroring to close a ticket in ServiceNow, under the **Mirrored XSOAR Ticket closure method** dropdown, select the ticket closing method,
```
Comment (Shellyber) [inline Packs/ServiceNow/Integrations/ServiceNowv2/README.md:41]: I'm sorry, shouldn't this one numbers 5 and 6 have the description revered? 
To enable mirroring to close incidents in Xsoar its what you've added
And the second one is closing the ticket (meaning mirror out)?

File: Packs/ServiceNow/Integrations/ServiceNowv2/README.md [hunk 6]
```diff
@@ -197,7 +195,7 @@ These tags work only for mirroring comments, work notes, and files from Cortex X
 15. Configure any **Custom Fields to Mirror**. These must start with "u_". This is available for ServiceNow v2 version 2.2.10 and later.  
   **Note:**  
   To enable mirroring custom fields, make a copy of the incoming and outgoing mappers and add the custom fields to the copies (see STEP 2 and STEP 3). Select these copies in the integration instance **Mapper (incoming)** and **Mapper (outgoing)** settings.
-16. To enable mirroring when closing an incident or ticket in Cortex XSOAR and ServiceNow, select **Close Mirrored XSOAR Incident** and **Close Mirrored ServiceNow Ticket** respectively.
+16. To enable mirroring when closing an incident or ticket in Cortex XSOAR and ServiceNow, select the *closed* option from the **Mirrored XSOAR Ticket closure method** dropdown and **Close Mirrored ServiceNow Ticket** respectively.
```
Comment (Shellyber) [inline Packs/ServiceNow/Integrations/ServiceNowv2/README.md:198]:
```suggestion
16. To enable mirroring when closing an incident or ticket in Cortex XSOAR and ServiceNow, select the *closed* option from the **Mirrored XSOAR Ticket closure method** dropdown and **Mirrored ServiceNow Ticket closure method** respectively.
```

File: Packs/ServiceNow/Integrations/ServiceNowv2/README.md [hunk 7]
```diff
@@ -300,7 +298,7 @@ These tags work only for mirroring comments from Cortex XSOAR to ServiceNow.
 17. Configure any **Custom Fields to Mirror**. These must start with "u_". This is available for ServiceNow v2 version 2.2.10 and later.  
   **Note:**  
   To enable mirroring custom fields, make a copy of the incoming and outgoing mappers and add the custom fields to the copies (see STEP 2 and STEP 3). Select these copies in the integration instance **Mapper (incoming)** and **Mapper (outgoing)** settings.
-18. To enable mirroring when closing an incident or ticket in Cortex XSOAR and ServiceNow, select **Close Mirrored XSOAR Incident** and **Close Mirrored ServiceNow Ticket** respectively.
+18. To enable mirroring when closing an incident or ticket in Cortex XSOAR and ServiceNow, select the *closed* option from the **Mirrored XSOAR Ticket closure method** dropdown **Close Mirrored ServiceNow Ticket** respectively.
```
Comment (Shellyber) [inline Packs/ServiceNow/Integrations/ServiceNowv2/README.md:301]: Same as above

File: Packs/ServiceNow/Integrations/ServiceNowv2/README.md [hunk 8]
```diff
@@ -2508,7 +2506,7 @@ The following are tips for handling issues with mirroring incidents between Serv
...
...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
```
Comment (Shellyber) [inline Packs/ServiceNow/Integrations/ServiceNowv2/README.md:2509]: Same as above, just need to fix the param naming

File: Packs/ServiceNow/ReleaseNotes/2_5_7.md [hunk 9]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### ServiceNow v2
+- Added the *close_incident_multiple_options* integration param to replace *close_incident* integration param. Now it is possible to change the incident-closing method. The options are 'closed' or 'resolved'.
```
Comment (Shellyber) [inline Packs/ServiceNow/ReleaseNotes/2_5_7.md:4]:
```suggestion
- Added the *Mirrored XSOAR Ticket closure method* integration param to replace *close_incident* integration param. Now it is possible to change the incident-closing method. The options are 'closed' or 'resolved'.
```

**Review by Shellyber — COMMENTED** [review]
Nice one!
See my last notes regarding the docs

File: Packs/ServiceNow/Integrations/ServiceNowv2/README.md (hunk 5 above)
Comment (samuelFain) [reply Packs/ServiceNow/Integrations/ServiceNowv2/README.md:41]: You are correct, i've reversed the descriptions.

File: Packs/ServiceNow/Integrations/ServiceNowv2/README.md (hunk 6 above)
Comment (samuelFain) [reply Packs/ServiceNow/Integrations/ServiceNowv2/README.md:198]: Done.

File: Packs/ServiceNow/Integrations/ServiceNowv2/README.md (hunk 7 above)
Comment (samuelFain) [reply Packs/ServiceNow/Integrations/ServiceNowv2/README.md:301]: Done.

File: Packs/ServiceNow/Integrations/ServiceNowv2/README.md (hunk 8 above)
Comment (samuelFain) [reply Packs/ServiceNow/Integrations/ServiceNowv2/README.md:2509]: Done.

File: Packs/ServiceNow/ReleaseNotes/2_5_7.md (hunk 9 above)
Comment (samuelFain) [reply Packs/ServiceNow/ReleaseNotes/2_5_7.md]: Done.

**Review by JasBeilin — APPROVED** [review]

**Pull request merged**